extensions/procfs/ProcessStat.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ProcessStat.h"
+#include <sstream>
+
+namespace org::apache::nifi::minifi::extensions::procfs {
+
+std::optional<ProcessStatData> ProcessStatData::parseProcessStatFile(std::istream& stat_file) {
+  std::stringstream stat_stream;
+  copy(std::istreambuf_iterator<char>(stat_file),
+       std::istreambuf_iterator<char>(),
+       std::ostreambuf_iterator<char>(stat_stream));
+
+  ProcessStatData process_stat_data;
+  stat_stream >> process_stat_data.pid_;
+
+  std::string stat_stream_string = stat_stream.str();
+  size_t comm_start = stat_stream_string.find_first_of('(');
+  size_t comm_end = stat_stream_string.find_last_of(')');
+  if (comm_start == std::string::npos  || comm_end == std::string::npos)
+    return std::nullopt;
+  process_stat_data.comm_ = stat_stream_string.substr(comm_start+1, comm_end-comm_start-1);
+  stat_stream.seekg(comm_end+2);
```

Comment:
why `comm_end + 2`? is there a whitespace we want to skip after the `)`?

Reply:
yeah there is a mandatory space after the `)`, but you are right this would work just as good with `comm_end+1` (because istream skips the whitespaces)
Should I change it?

Reply:
we can keep it, we could add a comment here